cmd/kubelet/app/server.go

```diff
@@ -986,16 +986,20 @@ func setContentTypeForClient(cfg *restclient.Config, contentType string) {
 //   2 Kubelet binary
 //   3 Standalone 'kubernetes' binary
 // Eventually, #2 will be replaced with instances of #3
-func RunKubelet(kubeServer *options.KubeletServer, kubeDeps *kubelet.Dependencies, runOnce bool) error {
-	hostname, err := nodeutil.GetHostname(kubeServer.HostnameOverride)
-	if err != nil {
-		return err
-	}
-	// Query the cloud provider for our node name, default to hostname if kubeDeps.Cloud == nil
-	nodeName, err := getNodeName(kubeDeps.Cloud, hostname)
-	if err != nil {
-		return err
+func RunKubelet(kubeServer *options.KubeletServer, kubeDeps *kubelet.Dependencies, runOnce bool, nodeName types.NodeName) error {
+
+	// If nodeName is empty , Assignment again
```

Comment:
Instead of this comment, can we use a descriptive variable name? I.e.

```
needCalculateNodeName := nodeName == ""  // Or `nodeName == nil... see my other comment
if needCalculateNodeName {
...
}
```